pkg/controller/buildrun/generate_taskrun_test.go

```diff
@@ -119,76 +132,80 @@ func TestGenerateTaskSpec(t *testing.T) {
-					corev1.ResourceCPU:    resource.MustParse("500m"),
-					corev1.ResourceMemory: resource.MustParse("1Gi"),
-				},
-			}
+				expectedCommandOrArg = []string{
+					"buildah", "bud", "--tls-verify=false", "--layers", "-f", fmt.Sprintf("$(inputs.params.%s)", "DOCKERFILE"), "-t", "$(outputs.resources.image.url)", fmt.Sprintf("$(inputs.params.%s)", "PATH_CONTEXT"),
+				}
 
-			assert.True(t, reflect.DeepEqual(err, nil))
+				expectedResourceOrArg = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("500m"),
+						corev1.ResourceMemory: resource.MustParse("1Gi"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("500m"),
+						corev1.ResourceMemory: resource.MustParse("1Gi"),
+					},
+				}
+			})
 
-			// ensure IMAGE is replaced by builder image when needed.
-			assert.Equal(t, fmt.Sprintf("$(inputs.params.%s)", inputParamBuilderImage), got.Steps[0].Container.Image)
+			JustBeforeEach(func() {
+				got, err = buildrunCtl.GenerateTaskSpec(build, buildRun, buildStrategy.Spec.BuildSteps)
+				Expect(err).To(BeNil())
+			})
 
-			// ensure command replacements happen when needed
-			assert.True(t, reflect.DeepEqual(got.Steps[0].Container.Command, expectedCommandOrArg))
+			It("should ensure IMAGE is replaced by builder image when needed.", func() {
+				Expect(got.Steps[0].Container.Image).To(Equal(fmt.Sprintf("$(inputs.params.%s)", "BUILDER_IMAGE")))
+			})
 
-			// ensure resource replacements happen when needed
-			assert.True(t, reflect.DeepEqual(got.Steps[0].Container.Resources, expectedResourceOrArg))
+			It("should ensure command replacements happen when needed", func() {
+				Expect(got.Steps[0].Container.Command).To(Equal(expectedCommandOrArg))
+			})
 
-			// ensure arg replacements happen when needed.
-			assert.True(t, reflect.DeepEqual(expectedCommandOrArg, got.Steps[0].Container.Args))
+			It("should ensure resource replacements happen when needed", func() {
+				Expect(got.Steps[0].Container.Resources).To(Equal(expectedResourceOrArg))
+			})
 
-			// Ensure top level volumes are populated.
-			assert.Equal(t, 2, len(got.Volumes))
-		})
-	}
-}
+			It("should ensure arg replacements happen when needed", func() {
+				Expect(got.Steps[0].Container.Args).To(Equal(expectedCommandOrArg))
+			})
 
-func TestGenerateTaskRun(t *testing.T) {
+			It("should ensure top level volumes are populated", func() {
+				Expect(len(got.Volumes)).To(Equal(2))
+			})
+		})
+	})
 
-	namespace := "build-test"
-	dockerfile := "Dockerfile"
-	contextDir := "src"
-	revision := "master"
-	builderImage := buildv1alpha1.Image{
-		ImageURL: "heroku/buildpacks:18",
-	}
-	clustertBuildStrategy := buildv1alpha1.ClusterBuildStrategyKind
-	outputPath := "image-registry.openshift-image-registry.svc:5000/example/buildpacks-app"
-	serviceAccountName := buildpacks + "-serviceaccount"
+	Describe("Generate the TaskRun", func() {
+		var (
```

Comment:
any reason why you define this set of variables per `Describe`, basically 2 times. Can they go at the very top on the first `var` block?

Also, in go you can always define similar types in one line, for example:
```
buildah, buildpacks, url, dockerfile string
```

Reply:
@qu1queee should I put all variables on the very top first `var` block ? Or just those were defined 2 times ?

Reply:
the ones that are define 2 times